Assets/Scripts/CartesianTiles/WFSGeoJSONTileDataLayer.cs

```diff
@@ -0,0 +1,144 @@
+
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Networking;
+using Newtonsoft.Json;
+using GeoJSON.Net.Feature;
+using Netherlands3D.Twin.Layers;
+
+namespace Netherlands3D.CartesianTiles
+{
+    /// <summary>
+    /// A custom CartesianTile layer that uses the cartesian tiling system to 'stream' parts of 
+    /// a WFS service to the client using the 'bbox' parameter.
+    /// The Twin GeoJSONLayer is used to render the GeoJSON data.
+    /// </summary>
+	public class WFSGeoJSONTileDataLayer : Layer
+	{		
+        private TileHandler tileHandler;
+        private string wfsUrl = "";
+        public string WfsUrl { get => wfsUrl; set => wfsUrl = value; }
+
+        private GeoJSONLayer geoJSONLayer;
+        public GeoJSONLayer GeoJSONLayer { 
+			get => geoJSONLayer; 
+			set
+			{
+				if(geoJSONLayer != null)
+					geoJSONLayer.ReferencedProxy.LayerDestroyed.RemoveListener(OnGeoJSONLayerDestroyed);
+
+				geoJSONLayer = value;
+				geoJSONLayer.ReferencedProxy.LayerDestroyed.AddListener(OnGeoJSONLayerDestroyed);
+			}
+		}
+
+        private void Awake() {
+			//Make sure Datasets at least has one item
+			if(Datasets.Count == 0)
+			{
+				var baseDataset = new DataSet()
+				{
+					maximumDistance = 3000,
+					maximumDistanceSquared = 1000 * 1000
+				};
+				Datasets.Add(baseDataset);
+			}
+
+			StartCoroutine(FindTileHandler());
+        }
+
+		private IEnumerator FindTileHandler()
+		{
+			yield return null;
+
+			//Find a required TileHandler in our parent, or else in the scene
+            tileHandler = GetComponentInParent<TileHandler>();
+            
+            if(!tileHandler)
+                tileHandler = FindAnyObjectByType<TileHandler>();
+
+            if(tileHandler){
+                tileHandler.AddLayer(this);
+                yield break;
+            }
+
+            Debug.LogError("No TileHandler found.", gameObject);
+		}
+
+		public override void HandleTile(TileChange tileChange, System.Action<TileChange> callback = null)
```

Comment:
Not for now, but I see this code is copied from the original layer, a nice refactoring would be if the Create, Upgrade, Downgrade and Remove action handlers were protected methods. This will help with readability but in this case you would only have to override the Create and Remove action, invoke their base and if we ever were to change the LOD handling, then we do not need to seek out the children.

Reply:
New ticket for seperating tiling system from CartesianTiles, so it can be used as base for WFS/WMS tiles without having dependency on CartesianTiles+Layers